sign/api/api.go

```diff
@@ -0,0 +1,33 @@
+// Package api contains a register of signature algorithms.
+package api
+
+import (
+	"github.com/cloudflare/circl/sign"
+	"github.com/cloudflare/circl/sign/ed25519"
+	"github.com/cloudflare/circl/sign/ed448"
+	"github.com/cloudflare/circl/sign/eddilithium3"
+	"github.com/cloudflare/circl/sign/eddilithium4"
+)
+
+var allSchemes [sign.SchemeCount]sign.Scheme
+var allSchemeNames map[string]sign.Scheme
+
+func init() {
+	allSchemeNames = make(map[string]sign.Scheme)
+	register(ed25519.Scheme)
+	register(ed448.Scheme)
+	register(eddilithium3.Scheme)
+	register(eddilithium4.Scheme)
+}
+
+func register(s sign.Scheme) {
+	allSchemes[s.ID()] = s
+	allSchemeNames[s.Name()] = s
+}
+
+// SchemeByName returns the scheme with the given name and nil if it is not
+// supported.
+func SchemeByName(name string) sign.Scheme { return allSchemeNames[name] }
+
+// AllSchemes returns all signature schemes supported.
+func AllSchemes() []sign.Scheme { a := allSchemes; return a[:] }
```

Comment:
This allows the caller to manipulate the list. That's dangerous.

Reply:
`a` is a copy, not a reference to the register

Reply:
Yes, you're right. I didn't read it correctly.